persistence/persistence-mybatis/src/main/resources/org/cbioportal/persistence/mybatis/MutationMapper.xml

```diff
@@ -151,6 +152,61 @@
         </where>
     </sql>
 
+    <sql id="whereInMultipleMolecularProfilesByAnnotation">
+        <where>
+            <if test="sampleIds != null and !sampleIds.isEmpty()">
+                mutation.SAMPLE_ID IN (
+                SELECT sample.INTERNAL_ID from sample
+                INNER JOIN patient ON sample.PATIENT_ID = patient.INTERNAL_ID
+                INNER JOIN genetic_profile ON patient.CANCER_STUDY_ID = genetic_profile.CANCER_STUDY_ID
+                WHERE
+                <if test="@java.util.Arrays@stream(molecularProfileIds.toArray()).distinct().count() == 1">
+                    genetic_profile.STABLE_ID = #{molecularProfileIds[0]} AND
+                    sample.STABLE_ID IN
+                    <foreach item="item" collection="sampleIds" open="(" separator="," close=")">
+                        #{item}
+                    </foreach>
+                </if>
+                <if test="@java.util.Arrays@stream(molecularProfileIds.toArray()).distinct().count() > 1">
+                    (sample.STABLE_ID, genetic_profile.STABLE_ID) IN
+                    <foreach index="i" collection="sampleIds" open="(" separator="," close=")">
+                        (#{sampleIds[${i}]}, #{molecularProfileIds[${i}]})
+                    </foreach>
+                    AND genetic_profile.STABLE_ID IN
+                    <foreach item="item" collection="molecularProfileIds" open="(" separator="," close=")">
+                        #{item}
+                    </foreach>
+                </if>
+                AND sample.INTERNAL_ID = mutation.SAMPLE_ID
+                AND genetic_profile.GENETIC_PROFILE_ID = mutation.GENETIC_PROFILE_ID)
+            </if>
+            <if test="sampleIds == null">
+                genetic_profile.STABLE_ID IN
+                <foreach item="item" collection="molecularProfileIds" open="(" separator="," close=")">
+                    #{item}
+                </foreach>
+            </if>
+            <if test="filters != null and !filters.isEmpty()">
+                <foreach item="filter" collection="filters">
+                    AND JSON_EXTRACT(mutation.annotation, #{filter.path}) in
```

Comment:
Here is the logic where the AnnotationFilter is applied ... the JSON is extracted and we are reading out (it appears) a single field ... and looking for matches to a set of supplied values